org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -227,23 +245,51 @@ public void scanFinished() throws ScanningException {
 		// Note: we don't use scanFinally as that is called after the nexus file is closed.
 		final Dataset scanFinishedDataset = DatasetFactory.createFromObject(IntegerDataset.class, 1, null);
 		try {
-			this.scanFinished.setSlice(null, scanFinishedDataset,
+			this.scanFinishedDataset.setSlice(null, scanFinishedDataset,
 					new int[] { 0 }, new int[] { 1 }, new int[] { 1 });
 		} catch (Exception e) {
 			throw new ScanningException("Could not write scanFinished to NeXus file", e);
 		}
 		
-		Duration scanDurationObj = Duration.between(scanStartTime, Instant.now());
-		String scanDurationStr = durationInMillisToString(scanDurationObj);
+		Duration scanDuration = Duration.between(scanStartTime, Instant.now());
+		String scanDurationStr = durationInMillisToString(scanDuration);
 		logger.info("Scan finished in " + scanDurationStr);
 		
 		final Dataset scanDurationDataset = DatasetFactory.createFromObject(scanDurationStr);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "scanDurationDataset" which hides the field declared at line 97. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)